apps/meteor/app/lib/server/functions/isTheLastMessage.ts

```diff
@@ -1,7 +1,7 @@
-import type { IMessage, IRoom } from '@rocket.chat/core-typings';
+import type { AtLeast, IMessage, IRoom } from '@rocket.chat/core-typings';
 
 import { settings } from '../../../settings/server';
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export const isTheLastMessage = (room: IRoom, message: Pick<IMessage, '_id'>) =>
+export const isTheLastMessage = (room: AtLeast<IRoom, 'lastMessage'>, message: Pick<IMessage, '_id'>) =>
 	settings.get('Store_Last_Message') && (!room.lastMessage || room.lastMessage._id === message._id);
```

Comment:
I know we didn't change this, but this looks wrong, the settings.get condition should not be here, the function is asking if this is the last message and mentions nothing about it being enabled...

Reply:
There's a couple of unrelated places using this func and assuming the setting check it's inside so i'll create a task for changing it later.